enterprise_subsidy/apps/api/v2/views/transaction.py

```diff
@@ -0,0 +1,202 @@
+from edx_rbac.decorators import permission_required
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from openedx_ledger.models import LedgerLockAttemptFailed, Transaction
+from rest_framework import generics, permissions, status
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.exceptions import APIException, NotFound, PermissionDenied, Throttled
+
+from enterprise_subsidy.apps.api.filters import TransactionAdminFilterSet
+from enterprise_subsidy.apps.api.utils import get_subsidy_customer_uuid_from_view
+from enterprise_subsidy.apps.api.v1.serializers import (
+    TransactionCreationError,
+    TransactionCreationRequestSerializer,
+    TransactionSerializer
+)
+from enterprise_subsidy.apps.subsidy.api import get_subsidy_by_uuid
+from enterprise_subsidy.apps.subsidy.constants import (
+    PERMISSION_CAN_CREATE_TRANSACTIONS,
+    PERMISSION_CAN_READ_ADMIN_TRANSACTIONS,
+    PERMISSION_CAN_READ_TRANSACTIONS
+)
+from enterprise_subsidy.apps.subsidy.models import ContentNotFoundForCustomerException, Subsidy
+
+logger = logging.getLogger(__name__)
+
+
+class TransactionBaseViewMixin:
+    """
+    Base view mixin that defines default authentication and permission classes;
+    a subsidy-scoped Transaction queryset; and a default Transaction serializer.
+    """
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = TransactionSerializer
+    queryset = Transaction.objects.all()
+
+    @property
+    def subsidy_uuid(self):
+        """
+        Returns the requested ``subsidy_uuid`` path parameter.
+        """
+        return self.kwargs.get('subsidy_uuid')
+
+    @cached_property
+    def subsidy(self):
+        """
+        Returns the Subsidy instance from the requested ``subsidy_uuid``.
+        """
+        return get_subsidy_by_uuid(self.subsidy_uuid, should_raise=True)
+
+    def get_queryset(self):
+        """
+        A base queryset that selects all transaction records (along with their
+        associated ledger and subsidy) for the requested ``subsidy_uuid``.
+        """
+        return Transaction.objects.select_related(
+            'ledger',
+            'ledger__subsidy',
+        ).filter(
+            ledger__subsidy=self.subsidy
+        )
+
+
+# https://drf-spectacular.readthedocs.io/en/latest/faq.html#using-extend-schema-on-apiview-has-no-effect
+# @extend_schema needs to be applied to the entrypoint method of the view.
+# For views derived from Viewset, these are methods like retrieve, list, create.
+# For APIView based views, these are get, post, create.
+# ListAPIView does in fact have a list method (via mixin), but the actual entrypoint
+# is still the get method, and the list call is proxied through the entrypoint.
+# Similarly for post/create.
+@extend_schema_view(
+    get=extend_schema(
```

Comment:
Nit: I feel like the code read faster if we put extend_schema on the view methods directly, otherwise you have to do a little bit of mental gymnastics to connect get -> list and create -> post